src/Sankey.js

```diff
@@ -0,0 +1,201 @@
+export default class Sankey extends Viz {
+
+  /**
+      @memberof Sankey
+      @desc Invoked when creating a new class instance, and sets any default parameters.
+      @private
+  */
+  constructor() {
+    super();
+    this._nodeId = false;
+    this._links = accessor("links");
+    this._noDataMessage = false;
+    this._nodes = accessor("nodes");
+    this._nodeWidth = 30;
+    this._sankey = sankey();
+    this._shapeConfig = assign(this._shapeConfig, {
+      labelConfig: {
+        duration: 0,
+        fontMin: 1,
+        fontResize: true,
+        labelPadding: 0,
+        textAnchor: "middle",
+        verticalAlign: "middle"
+      },
+      Path: {
+        fill: "none",
+        label: false,
+        stroke: "#dbdbdb",
+        strokeWidth: 20,
+        strokeOpacity: 0.4
+      }
+    });
+  }
+
+  /**
+      Extends the draw behavior of the abstract Viz class.
+      @private
+  */
+  _draw(callback) {
+    super._draw(callback);
+
+    const height = this._height - this._margin.top - this._margin.bottom,
+          width = this._width - this._margin.left - this._margin.right;
+
+    const path = sankeyLinkHorizontal();
+
+    const nodes = this._nodes;
+
+    const nodeLookup = this._nodeLookup = this._nodes.reduce((obj, d, key) => {
+      obj[d[this._nodeId || "id"]] = key;
+      return obj;
+    }, {});
+
+    const links = this._links.map(link => {
+      const check = ["source", "target"];
+
+      const linkLookup = check.reduce((result, item) => {
+        result[item] =
+          typeof link[item] === "number" ? link[item] : nodeLookup[link[item]];
+        return result;
+      }, {});
+
+      return {
+        source: linkLookup.source,
+        target: linkLookup.target,
+        value: link.value
+      };
+    });
+
+    const transform = `translate(${this._margin.left}, ${this._margin.top})`;
+
+    this._sankey
+      .nodeWidth(this._nodeWidth)
+      .nodes(nodes)
+      .links(links)
+      .size([width, height])();
+
+    this._shapes.push(
+      new Path()
+        .config(this._shapeConfig)
+        .config(this._shapeConfig.Path)
+        .data(links)
+        .d(path)
+        .select(
+          elem("g.d3plus-Links", {
+            parent: this._select,
+            enter: {transform},
+            update: {transform}
+          }).node()
+        )
+        .config(assign(this._tooltip, false))
+        .config(
+          configPrep.bind(this)(
+            assign(this._shapeConfig, {
+              Path: {
+                strokeWidth: d =>
+                  Math.max(
+                    1,
+                    Math.abs(d.source.y1 - d.source.y0) *
+                      (d.value / d.source.value) -
+                      20
+                  )
+              }
+            }),
+            "shape",
+            "Path"
+          )
+        )
+        .render()
+    );
+
+    this._shapes.push(
+      new Rect()
+        .config(this._shapeConfig)
+        .config(this._shapeConfig.Path)
```

Comment:
are these two initial `config` lines are necessary? you're applying the user-defined shapeConfig, which gets applied again at the end of the chain, and then you're applying the Path config, which doesn't apply to Rect